src/Controller/CRUDController.php

```diff
@@ -402,7 +401,12 @@ public function batchAction(Request $request): Response
             throw new BadRequestParamHttpException('data', 'string', $encodedData);
         }
 
-        $data = json_decode($encodedData, true);
+        try {
+            $data = json_decode($encodedData, true, 512, \JSON_THROW_ON_ERROR);
+        } catch (\JsonException $exception) {
+            $data = null;
```

Comment:
I wonder if we should throw a BadRequestHttpException

Reply:
Not sure how to fix it, since I think we are currently relying on a null return (when \JSON_THROW_ON_ERROR is added then it is an exception) to go through the else part: https://github.com/sonata-project/SonataAdminBundle/blob/4.x/src/Controller/CRUDController.php#L411

Reply:
We're doing `$encodedData = $request->get('data', '');`.

I would say that
- Empty string `''` is the else part
- String decoded as array is the if
- String which cannot be decoded throw a BadRequestHttpException

Reply:
If you try to decode with `''` you will get an exception (that's what happens on some of our test suite)

Reply:
IMO we can change to:
1. empty string -> null
2. If data is null go to the else
3. Otherwise go to the if and try to decode the string, throw exception if it does not decode well

Reply:
I thought about a `!== ''` check before the decode but

> IMO we can change to:
> empty string -> null
> If data is null go to the else
> Otherwise go to the if and try to decode the string, throw exception if it does not decode well

seems better indeed

Reply:
Fixed.